src/controllers/controllermanager.cpp

```diff
@@ -205,6 +217,10 @@ QList<Controller*> ControllerManager::getControllerList(bool bOutputDevices, boo
     return filteredDeviceList;
 }
 
+KeyboardController* ControllerManager::getKeyboardController() {
```

Comment:
It is unsafe to hand out a pointer that might be deleted during the lifetime of its owner. As a caller I would expect that the pointer exists at least until ControllerManager is destroyed. Otherwise you should consider to use a QSharedPointer for m_pKeyboardController.